Don't drop groupby hash table for 'tick lifetimes

zzlk opened the pull request.

zzlk commented:
We talked earlier about this maybe breaking with groupby and loops but conclude it's probably not more broken than it currently is to why not merge it?

MingweiSamuel reviewed (approved):
Same code from this afternoon?

zzlk commented:
> Same code from this afternoon?

ya, I cleaned it up a bit, hid the check_type fn inside a scope more consistently and fixed some indenting but otherwise the same.
